'template with C linkage' in imap/xapian_wrap.cpp

Adding `#include <unicode/ucnv.h>` outside the extern "C" block lets it get past that. I'm not sure if that's a valid fix though.

```
In file included from /usr/include/c++/8.3.0/memory:64,
                 from /usr/include/unicode/localpointer.h:45,
                 from /usr/include/unicode/uenum.h:23,
                 from /usr/include/unicode/ucnv.h:53,
                 from ./lib/charset.h:61,
                 from imap/xapian_wrap.cpp:17:
/usr/include/c++/8.3.0/bits/stl_construct.h:72:3: error: template with C linkage
   template<typename _T1, typename... _Args>
   ^~~~~~~~
imap/xapian_wrap.cpp:11:1: note: 'extern "C"' linkage started here
 extern "C" {
 ^~~~~~~~~~
```

--- a/imap/xapian_wrap.cpp
+++ b/imap/xapian_wrap.cpp
@@ -4,7 +4,9 @@
 #include <string.h>
 #include <sys/types.h>
 #include <syslog.h>
+#ifdef HAVE_ICU
 #include <unicode/ucnv.h>
+#endif
 
 #include <fstream>
 #include <sstream>

--- a/imap/xapian_wrap.cpp
+++ b/imap/xapian_wrap.cpp
@@ -4,6 +4,7 @@
 #include <string.h>
 #include <sys/types.h>
 #include <syslog.h>
+#include <unicode/ucnv.h>
 
 #include <fstream>
 #include <sstream>

